modular_eros/code/datums/status_effects/pregnancy.dm

```diff
@@ -0,0 +1,238 @@
+	REMOVE_TRAIT(owner, TRAIT_PREGNANT, PREGNANCY_TRAIT)
+	UnregisterSignal(owner, COMSIG_MOB_DEATH)
+	if(ishuman(owner))
+		human_pragency_end(owner)
+	return ..()
+
+/datum/status_effect/pregnancy/proc/human_pragency_start(mob/living/carbon/human/gregnant)
+	//you need a womb even if male to get gragnet
+	if(!gregnant.getorganslot(ORGAN_SLOT_WOMB))
+		return FALSE
+	if(gregnant.client?.prefs)
+		oviposition = gregnant.client.prefs.oviposition
+		pregnancy_inflation = gregnant.client.prefs.pregnancy_inflation
+		if(father && ispath(baby_type, /mob/living/carbon/human))
+			INVOKE_ASYNC(src, .proc/progenitor_select)
+	if(pregnancy_inflation)
+		//give them a king ass ripper belly initially
+		var/obj/item/organ/genital/belly/belly = gregnant.getorganslot(ORGAN_SLOT_BELLY)
+		if(!belly)
+			belly = gregnant.give_genital(/obj/item/organ/genital/belly)
+		else
+			old_belly_size = belly.size
+		belly.size = max(belly.size, 1)
+		belly.update()
+	return TRUE
+
+/datum/status_effect/pregnancy/proc/human_pragency_end(mob/living/carbon/human/gregnant)
+	//get rid of king ass ripper belly
+	var/obj/item/organ/genital/belly/belly = gregnant.getorganslot(ORGAN_SLOT_BELLY)
+	if(belly && pregnancy_inflation)
+		belly.size = old_belly_size
+		belly.update()
+
+/datum/status_effect/pregnancy/proc/progenitor_select()
+	var/list/buttons = list("Like me!", "Like dad!")
+	var/fatherful_input = tgui_input_list(owner, "You're pregnant! Who is the baby gonna look like?", "Pregnancy", buttons, 1 MINUTES)
+	if(fatherful_input == "Like dad!")
+		like_mother_like_son = FALSE
+
+/datum/status_effect/pregnancy/proc/fetus_mortus()
+	SIGNAL_HANDLER
+
+	if(!QDELETED(owner) && isturf(owner.loc))
+		if(!oviposition)
+			new /obj/effect/gibspawner/generic(owner.loc)
+		else
+			new /obj/effect/decal/cleanable/egg_smudge(owner.loc)
+	owner.visible_message(span_danger("[owner] has a miscarriage!"), \
+						span_userdanger("Oh no! My baby is dead!"))
+	qdel(src)
+
+/datum/status_effect/pregnancy/proc/offer_control_to_babby(mob/living/babby, mob/living/mommy)
+	var/poll_message = "Do you want to play as [mommy]'s offspring?"
+	var/list/mob/candidates = pollCandidatesForMob(poll_message, ROLE_RESPAWN, null, FALSE, 120, babby)
+	if(!LAZYLEN(candidates))
+		babby.real_name = random_unique_name(babby.gender, FALSE)
+		babby.update_name()
+		return
+
+	var/mob/player = pick(candidates)
+	player.transfer_ckey(babby, TRUE)
+	var/mommy_name = "someone"
+	if(!QDELETED(mommy))
+		mommy_name = mommy.real_name
+
+	to_chat(babby, "You are the son (or daughter) of [mommy_name]!")
+	var/time = world.time
+	var/name = input(babby, "What will be your name? (You have a minute to decide!)", "Name yourself") as null|text
+	if(world.time - time >= 1 MINUTES)
+		to_chat(babby, "You took too long to decide a name and have been given a random name instead.")
+	if(!name || (world.time - time >= 1 MINUTES))
+		babby.real_name = random_unique_name(babby.gender, FALSE)
```

Comment:
Shouldn't the baby's surname be the same as the mother/father?

Reply:
Will change this so the mother decides the name, if possible, actually.